aptos-move/framework/move-stdlib/tests/mem_tests.move

```diff
@@ -0,0 +1,97 @@
+#[test_only]
+module std::mem_tests {
+    use std::vector;
+    use std::mem::{swap, replace};
+
+    #[test]
+    fun test_swap_ints() {
+        let a = 1;
+        let b = 2;
+        let v = vector[3, 4, 5, 6];
+
+        swap(&mut a, &mut b);
+        assert!(a == 2, 0);
+        assert!(b == 1, 1);
+
+        swap(&mut a, vector::borrow_mut(&mut v, 0));
+        assert!(a == 3, 0);
+        assert!(vector::borrow(&v, 0) == &2, 1);
+
+        swap(vector::borrow_mut(&mut v, 2), &mut a);
+        assert!(a == 5, 0);
+        assert!(vector::borrow(&v, 2) == &3, 1);
+    }
+
+    #[test]
+    fun test_replace_ints() {
+        let a = 1;
+        let b = 2;
+
+        assert!(replace(&mut a, b) == 1, 0);
+        assert!(a == 2, 1);
+    }
+
+    #[test_only]
+    struct SomeStruct has drop, key {
```

Comment:
Docstring in `mem.move` says we do this for non-copyable/non-droppable types, but here there is drop?